### ISSUE TITLE
RyuJIT x86: NYI: "Unimplemented type for Lowering of cast to TYP_LONG"

### ISSUE THREAD

**Issue by adiaaida** [dotnet/coreclr#4175]
This NYI:

NYI: "Unimplemented type for Lowering of cast to TYP_LONG"
File: coreclr\src\jit\lower.cpp Line: 350

Is hit by these tests:

JIT\Methodical\xxobj\sizeof\_dbgsizeof64\_dbgsizeof64.cmd
JIT\Methodical\xxobj\sizeof\_relsizeof64\_relsizeof64.cmd
JIT\Regression\CLR-x86-JIT\V1-M12-Beta2\b71120\b71120\b71120.cmd
JIT\Methodical\flowgraph\dev10_bug679008\dependentlifetimes\dependentlifetimes.cmd

**Comment by mikedn** [comment]
The correct paths for the first 2 tests are:
```
JIT\Methodical\xxobj\sizeof\_dbgsizeof64\_dbgsizeof64.cmd
JIT\Methodical\xxobj\sizeof\_relsizeof64\_relsizeof64.cmd
```

**Comment by mikedn** [comment]
Looks like b71120 uses a byte->long cast and the rest use byref->long casts.

**Comment by mikedn** [comment]
Fixed by #4986
